core/modules/system/system.install

```diff
@@ -98,6 +100,21 @@ function system_requirements($phase) {
     $requirements['php_register_globals']['value'] = $t('Disabled');
   }
 
+  // Test PHP max_input_vars setting.
+  $max_input_vars = ini_get('max_input_vars');
+  $requirements['php_max_input_vars'] = array(
+    'title' => $t('PHP form input max'),
+    'value' => $t("PHP limits the number of form values to @value items.", array('@value' => number_format($max_input_vars, 0, '.', ','))),
+  );
+  // 1000 is the default value, and indicates potential issues.
+  if ($max_input_vars == 1000) {
+    $requirements['php_max_input_vars']['description'] = $t('When set too low, PHP\'s <code>max_input_vars</code> setting can cause silent failures on large forms. Use caution when using the permissions page, ordering large menus, arraging large vocabularies, or editing content with lots of fields.');
+    $requirements['php_max_input_vars']['severity'] = REQUIREMENT_ERROR;
+  }
+  else {
```

Comment:
This should be an `elseif`, and also check for more than a specific number. Something like:

```suggestion
  elseif ($max_input_vars >= 10000) {
```

Not sure what the exact number should be, but because the previous `if` statement checks for `== 1000`, this means that as is now, this condition will validate as TRUE even with a limit set to 1, or 50, or 100, telling people that that is sufficient.

Reply:
good catch.

Reply:
🙂